### DISCUSSION TITLE
How to configure debugging/codelldb

### DISCUSSION THREAD

**Discussion by FritsHoogland** [mrcjkb/rustaceanvim#159] [Q&A] [answered]
There is an example on how to configure codelldb for rustaceanvim on the website, but I can't figure out where to put this configuration so that nvim/Astronvim can pick it up. It is probably dead simple...

**Comment by mrcjkb** [comment, marked as answer]
Hey :wave: 

If you have `codelldb` installed (e.g. via a package manager or mason.nvim), this plugin should pick up the configuration for you, and you shouldn't need to configure anything manually.

If it doesn't, you can follow the [tutorial in the readme](https://github.com/mrcjkb/rustaceanvim?tab=readme-ov-file#using-codelldb-for-debugging).

The `vim.g.rustaceanvim` table can go anywhere in your Neovim config (e.g. `init.lua`).
The only place it should **not** go is a `ftplugin/rust.lua` file, because it would be initialized too late.

I don't use AstroNVim, and don't know if it's configured differently than regular nvim, so I can't provide direct support for it.
Afaik, it uses lazy.nvim, so you could also do this:

```lua
{
  'mrcjkb/rustaceanvim',
  init = function()
    vim.g.rustaceanvim = ...
  end,
}
```
